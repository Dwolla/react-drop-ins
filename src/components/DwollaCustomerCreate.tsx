import type { FC } from 'react';
import type { CreateCustomerProps } from '../types/component-props';

const DwollaCustomerCreate: FC<CreateCustomerProps> = (props) => {
  return <dwolla-customer-create {...props} />;
};

export default DwollaCustomerCreate;
