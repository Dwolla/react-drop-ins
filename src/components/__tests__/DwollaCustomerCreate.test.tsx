import TestRenderer from 'react-test-renderer';
import DwollaCustomerCreate from '../DwollaCustomerCreate';

test('DwollaCustomerCreate', () => {
  const tree = TestRenderer.create(<DwollaCustomerCreate />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('with optional props', () => {
  const tree = TestRenderer.create(
    <DwollaCustomerCreate
      correlationId="123abc"
      email="janedoe@email.com"
      firstName="Jane"
      lastName="Doe"
      ipAddress="255.255.255.255"
      privacy="www.yourprivacy.com"
      terms="www.yourterms.com"
      businessName="Jane Doe, LLC"
      isBusiness
      type="receive-only"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
