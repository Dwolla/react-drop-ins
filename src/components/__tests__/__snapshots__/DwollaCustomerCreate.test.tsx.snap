// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DwollaCustomerCreate 1`] = `<dwolla-customer-create />`;

exports[`with optional props 1`] = `
<dwolla-customer-create
  businessName="Jane Doe, LLC"
  correlationId="123abc"
  email="janedoe@email.com"
  firstName="Jane"
  ipAddress="255.255.255.255"
  isBusiness={true}
  lastName="Doe"
  privacy="www.yourprivacy.com"
  terms="www.yourterms.com"
  type="receive-only"
/>
`;
